controllers/flinkcluster/flinkcluster_updater.go

```diff
@@ -678,7 +678,8 @@ func (updater *ClusterStatusUpdater) deriveJobStatus(ctx context.Context) *v1bet
 			newJob.Name = observedFlinkJob.Name
 		}
 		tmpState := getFlinkJobDeploymentState(observedFlinkJob.State)
-		if observedSubmitter.job == nil || tmpState != v1beta1.JobStateSucceeded {
+		if observedSubmitter.job == nil ||
+			(observedSubmitter.job.Status.Failed < 1 && tmpState != v1beta1.JobStateSucceeded) {
```

Comment:
Hey @yolgun 
Are you trying to check if the submitter is still initialising here? 🤔 

Recently added a utility function for it [here](https://github.com/spotify/flink-on-k8s-operator/blob/master/controllers/flinkcluster/flinkcluster_util.go#L608)

Reply:
Hi @live-wire . At this PR, I want to check for `observedSubmitter.job.Status.Failed`. And if it is, it should fallthrough the next case, where the job state can be marked as failed.

Reply:
Yeah this makes sense. Nice find!